src/bin/nydus-image/core/bootstrap.rs

```diff
@@ -264,10 +264,13 @@ impl Bootstrap {
         }
 
         // dot & dotdot
-        node.dirents
-            .push((node.offset, OsString::from("."), libc::S_IFDIR));
-        node.dirents
-            .push((parent_offset, OsString::from(".."), libc::S_IFDIR));
+        #[allow(clippy::useless_conversion)]
```

Comment:
Maybe we can explain why we do this.

Reply:
Type of libc::S_IFDIR is u16 on macos, so it need a conversion; but compiler will report useless conversion on linux platform, so we add an allow annotation here.

Reply:
Thanks!

Reply:
> Type of libc::S_IFDIR is u16 on macos, so it need a conversion; but compiler will report useless conversion on linux platform, so we add an allow annotation here.

Please add it to the code comment, thanks!